### PR TITLE
identity: factor kvstore-related code out into sub-package

### PR DESCRIPTION
**Pull request by ianvernon**
This allows for consumers of identities to not import `pkg/kvstore`, as not all
consumers of identities allocate them.

Signed-off by: Ian Vernon <ian@cilium.io>

Related-to: #6101

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6154)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

File: pkg/identity/cache/cache.go [hunk 1]
```diff
@@ -12,26 +12,29 @@
...
 )
 
 var (
-	reservedIdentityCache = map[NumericIdentity]*Identity{}
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "identity-cache")
 )
 
-// IdentityCache is a cache of identity to labels mapping
-type IdentityCache map[NumericIdentity]labels.LabelArray
+// IdentityCacheIfc is a cache of identity to labels mapping
```
[the remark this answers is not in the document]
Comment (ianvernon) [reply pkg/identity/cache/cache.go]: Will be fixed in next upload.

File: pkg/identity/reserved.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package identity
+
+import "github.com/cilium/cilium/pkg/labels"
+
+var (
+	ReservedIdentityCache = map[NumericIdentity]*Identity{}
```
[the remark this answers is not in the document]
Comment (ianvernon) [reply pkg/identity/reserved.go:22]: Will be fixed in next upload.

File: pkg/identity/numericidentity.go [hunk 3]
```diff
@@ -206,7 +206,7 @@ var (
 		ReservedIdentityInit:      labels.IDNameInit,
 	}
 
-	wellKnown = wellKnownIdentities{}
+	WellKnown = wellKnownIdentities{}
```
[the remark this answers is not in the document]
Comment (ianvernon) [reply pkg/identity/numericidentity.go:210]: Will be fixed in next upload.

File: pkg/identity/cache/allocator.go [hunk 4]
```diff
@@ -51,7 +52,7 @@ func (gi globalIdentity) PutKey(v string) (allocator.AllocatorKey, error) {
 
 var (
 	setupOnce         sync.Once
-	identityAllocator *allocator.Allocator
+	IdentityAllocator *allocator.Allocator
```
[the remark this answers is not in the document]
Comment (ianvernon) [reply pkg/identity/cache/allocator.go:58]: Fixed in next upload.

File: pkg/identity/identity.go [hunk 5]
```diff
@@ -37,6 +37,10 @@ type Identity struct {
 	LabelArray labels.LabelArray `json:"-"`
 }
 
+type IdentityCacheIfc interface {
```
[the remark this answers is not in the document]
Comment (ianvernon) [reply pkg/identity/identity.go]: Removed in next upload.

**Comment by ianvernon** [comment]
test-me-please

**Review by jrajahalme — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Pull request merged**